lib/db_utils.pm

```diff
@@ -54,8 +54,11 @@ sub build_influx_query {
     C<data> should contain a hash containing the table name in Influx DB, the tags and the values to plot.
 =cut
 sub influxdb_query {
-    my ($url, $data, %args) = @_;
+    my ($url, $db, $data, %args) = @_;
     $args{quiet} //= 1;
     $data = build_influx_query($data);
-    assert_script_run(sprintf("curl -i -X POST '%s' --data-binary '%s'", $url, $data), quiet => $args{quiet});
+    my $cmd = sprintf("curl -i -X POST '%s/write?db=%s' --data-binary '%s'", $url, $db, $data);
+    record_info('curl', $cmd);
+    my $output = script_output($cmd, quiet => $args{quiet});
+    die("Something wrong happened while pushing results to Influx DB:\n$output") if ($output =~ /404/);
```

Comment:
```suggestion
    die("Fail to push data into $db:\n$output") if ($output =~ /404/);
```
no need to write too long message , also you just introduce flexibility to which DB but miss this in error message :)

Reply:
another thing - a lot of other things may happen while posting data with curl ( e.g. 50* errors , timeout issues etc. ) why we care only about 404 ? 
maybe we need just do it other way around ? I mean check that we get 200 ?

Reply:
> no need to write too long message , also you just introduce flexibility to which DB but miss this in error message :)

my bad I confused db name with db type :) but still vote for shorter message

Reply:
> another thing - a lot of other things may happen while posting data with curl ( e.g. 50* errors , timeout issues etc. ) why we care only about 404 ?
> maybe we need just do it other way around ? I mean check that we get 200 ?

good point. Actually, what we are expecting is something like:

```
HTTP/1.1 204 No Content
Content-Type: application/json
```

Reply:
What about 

    die("Fail to push data into Influx DB:\n$output") if ($output !~ /204 No Content/);

Reply:
+1

Reply:
I think I would also check the return code like:
```
my $cmd = sprintf("curl -X POST '%s/write?db=%s' --data-binary '%s' --write-out 'RETURN_CODE:%{http_code}\n'", $url, $db, $data);
my $output = script_output($cmd, quiet => $args{quiet});
my ($return_code) = $output =~ /RETURN_CODE:(\d+)/;
die (msg) unless ($return_code => 200 && $return_code < 300);
```
it's not tested...